lib/nats.js

```diff
@@ -999,7 +1052,8 @@ Client.prototype.processErr = function(s) {
     // except stale connection and permission errors
     var m = s ? s.toLowerCase() : '';
     if (m.indexOf(STALE_CONNECTION_ERR) !== -1) {
-        this.scheduleReconnect();
+        this.closeStream();
+        // this.scheduleReconnect();
```

Comment:
Remove this and maybe instead add comment above to say that `closeStream()` will trigger reconnect if allowed.